_ap/16/index.adoc

```diff
@@ -13,56 +13,131 @@ tags:
 
 ## Structure
 
-// Describe the pattern.
-//
-// For this metrics example we might link to existing guidlines on naming, augment them with extra rules 
-// which apply to services, explain how the API would allow for graceful API evolution without "flag days".
+### Metric Name
+The metric name should adhere to the format below:
 
-## Participants
+*<service-name>_<optional:sub-component>_<descriptor>_<optional:base-unit>_<optional:type|aggregate>*
 
-// What components and/or teams are directly involved
-//
-// For this metrics example this would include the customer and the control plane via which
-// the metrics are exposed
+* *service-name*: The metric name should be prefixed with the service name (e.g. `rhosak`)
+* *sub-component* (optional): The sub-component of the service that the metric measures.
+This should be used for logically categorizing the metrics. 
+The different sub-components of the service must be agreed on by the team (e.g. `server`, `producer`, `consumer`).
+* *descriptor*: Describes what is being measured.
+* *base-unit* (optional): Describes the unit used for measuring the metric (e.g. `seconds`, `bytes`). 
+This should be in plural form.
+See https://prometheus.io/docs/practices/naming/#base-units[Prometheus base units] for a list of units that can be used here.
+* *type|aggregate* (optional): Describes the type of or aggregate used for measuring the metric. (e.g. `total`, `rate<time>`, `info`, `bucket`)
 
-## Consequences
+Metric names must also adhere to the following:
+
+* Must match the regex `[a-z_][a-z0-9_]*`.
+* Each word should be separated by `_` (e.g. the descriptor `availablevolume` should be seperated into `available_volume`).
+
+### Label Name
+
+Label names must adhere to the following:
+
+* Must match the regex `[a-z_][a-z0-9_]*`.
+* Each word should be separated by `_` (e.g. the label name `brokername` should be seperated into `broker_name`).
+* Must have consistent names across all metrics. 
+Avoid using different label names in different metrics that describes the same feature.
+
+
+### Things to avoid
+
+* Label names should not be included in the metric name.
+This introduces redundancy and will cause confusion if the respective labels are aggregated away.
+* Use *:* in the metric or label name. This is reserved for recording rules only.
+Even if the underlying metric is a recording rule, this should be avoided as this is an implementation detail and should not be exposed to the end user.
+* Use any other cases apart from snake_case.
+* Expose underlying components in the metric or label name.
+For example, avoid including project names like 'strimzi' or 'kafka-exporter' or OpenShift/Kubernetes resources like 'namespaces' or 'projects'.
+
+### Deprecating Metrics
+The team's API deprecation policy must be followed when deprecating metrics. 
+
+The length of the deprecation period should be agreed upon by the team and applied for all metrics exposed in their API.
+
+
+### Examples
```

Comment:
Similar to [https://github.com/bf2fc6cc711aee1a0c2a/architecture/pull/64/files#r892526338](comment) above. Examples here were based on sub-components `server`, `producer`, `consumer`. Will change these once we agree what sub-components we'll have for rhosak.